app/forms/field_handlers/number_handler.py

```diff
@@ -84,6 +50,20 @@ def get_field_references(
             "maximum": maximum,
         }
 
+    @cached_property
+    def max_decimals(self) -> int:
+        return self.answer_schema.get("decimal_places", 0)
+
+    def get_field(self) -> Union[DecimalField, IntegerField]:
+        field_type = (
+            DecimalFieldWithSeparator
+            if self.max_decimals > 0
```

Comment:
Very minor but I'd suggest a property that evaluates this, with the same logic but called something like `is_decimal_field`

Reply:
Was existing, hence I didn't bother with any refactor here, but have changed while I'm here 👍